modules/dnn/src/model.cpp

```diff
@@ -401,4 +407,310 @@ void DetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
         CV_Error(Error::StsNotImplemented, "Unknown output layer type: \"" + lastLayer->type + "\"");
 }
 
+TextRecognitionModel::TextRecognitionModel(const String& model, const String& config)
+    : Model(model, config) {}
+
+TextRecognitionModel::TextRecognitionModel(const Net& network) : Model(network) {}
+
+void TextRecognitionModel::setVocabulary(const std::vector<String>& voc)
+{
+    vocabulary.clear();
+    for (uint i = 0; i < voc.size(); i++) {
+        vocabulary.push_back(voc.at(i));
+    }
```

Comment:
Please use `stl` - resize and insert

Reply:
We need to set the `vocabulary` with the input `voc`, so I think `assign` is a better choice than `insert`.
Please check it and let me know if it is not good. Thank you for your help.

Reply:
Just `vocabulary = voc;` should work fine.